### ISSUE TITLE
Investigate whether OR conditions can be supported.

### ISSUE THREAD

**Issue by derjust** [derjust/spring-data-dynamodb#4]
> Migration of https://github.com/michaellavelle/spring-data-dynamodb/issues/3

**Comment by fpoolev** [comment]
+1

**Comment by fpoolev** [comment]
After doing a little research, it looks like implementing the OR condition is dependent upon first implementing filter expressions (see #42) as the previous implementation path using setConditionalOperator is now considered a legacy parameter.

AWS Documentation on the Conditional Operator (such as in a DynamoDBQueryExpression):
> This is a legacy parameter, for backward compatibility. New applications should use FilterExpression instead. Do not combine legacy parameters and expression parameters in a single call; otherwise, DynamoDB will return a ValidationException exception.
http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html#ConditionalOperator

AWS Documentation on Filter Expressions:
http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/QueryAndScan.html#FilteringResults

I did not find anywhere in the code base in which the conditional operator for a query is explicitly set to AND. I'm assuming that AND is already supported because the conditional operator defaults to AND.

**Comment by derjust** [comment]
That's correct. #28 needs to go away first and would make things easier for sure in general
